hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/upgrade/UpgradeFinalizer.java

```diff
@@ -187,4 +186,15 @@ StatusAndMessages reportStatus(String upgradeClientId, boolean takeover)
    */
   void runPrefinalizeStateActions(Storage storage, T service)
       throws IOException;
+
+  /**
+   * Sets the Finalization Executor driver.
+   * @param executor FinalizationExecutor.
+   */
+  void setFinalizationExecutor(UpgradeFinalizationExecutor executor);
```

Comment:
Can we create overriden constructors to the finalizers (DN, SCM) that take in a custom executor? We can add one with a new executor parameter. If not passed in, the default executor will be used. In that case, we can remove the setter in the interface.

Reply:
Currently default constructor for DN SCM gets the default constructor only. The set method is only for the TestHDDSUpgrade.